Sources/LanguageServerProtocol/WorkspaceEdit.swift

```diff
@@ -14,9 +14,13 @@
 public struct WorkspaceEdit: Codable, Hashable, ResponseType {
 
   /// The edits to be applied to existing resources.
-  public var changes: [URL: [TextEdit]]?
+  public var changes: [String: [TextEdit]]?
 
   public init(changes: [URL: [TextEdit]]?) {
-    self.changes = changes
+    guard let changes = changes else {
```

Comment:
I haven't fully investigated the reason, but using `URL` directly here is tripping the VSCode extension. It causes `changes` to be parsed as an array instead of a dictionary:

```
[Trace - 10:16:58 PM] Received request 'workspace/applyEdit - (1)'.
Params: {
    "label": "Extract Method",
    "edit": {
        "changes": [
            "file:///file.swift",
            [
                {
                    "range": {
                        "start": {
                            "line": 20,
                            "character": 0
                        },
```

vs (String):

```
[Trace - 10:19:36 PM] Received request 'workspace/applyEdit - (1)'.
Params: {
    "label": "Extract Method",
    "edit": {
        "changes": {
            "file:///file.swift": [
                {
                    "range": {
                        "start": {
                            "line": 20,
                            "character": 0
                        },
```

Gonna look further into it, but I made the key a string meanwhile.

Reply:
Relevant thread on swift forums: https://forums.swift.org/t/using-rawrepresentable-string-and-int-keys-for-codable-dictionaries/26899